lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -146,384 +167,352 @@ class AwsCompileFunctions {
         );
       }
 
-      if (this.serverless.service.package.deploymentBucket) {
-        functionResource.Properties.Code.S3Bucket = this.serverless.service.package.deploymentBucket;
-      }
-
-      const s3Folder = this.serverless.service.package.artifactDirectoryName;
-      const s3FileName = artifactFilePath.split(path.sep).pop();
-      functionResource.Properties.Code.S3Key = `${s3Folder}/${s3FileName}`;
-
-      if (!functionObject.handler) {
-        const errorMessage = [
-          `Missing "handler" property in function "${functionName}".`,
-          ' Please make sure you point to the correct lambda handler.',
-          ' For example: handler.hello.',
-          ' Please check the docs for more info',
-        ].join('');
-        throw new this.serverless.classes.Error(errorMessage);
-      }
+      functionObject.runtime = this.provider.getRuntime(functionObject.runtime);
+      functionResource.Properties.Handler = functionObject.handler;
+      functionResource.Properties.Code.S3Bucket = this.serverless.service.package.deploymentBucket
+        ? this.serverless.service.package.deploymentBucket
+        : { Ref: 'ServerlessDeploymentBucket' };
```

Comment:
Is this ref guaranteed to resolve? I noticed that it's not in the previous version of this code, and `functionResource.Properties.Code.S3Bucket` would be left `null` if no bucket was present.

Reply:
_This code was just moved to `if` clause, so technically this PR doesn't introduce any changes in how `Code.S3Bucket` is configured_

By default we use our bucket which is created with a stack (and then `Ref` reference is assigned), and in other case user provides reference to bucket created externally (and format of bucket reference, as put on `package.deploymentBucket` is guarded by schema based validation)

Reply:
Okay, I see. Somehow I did not see that part during my initial review